app/models/product.rb

```diff
@@ -20,6 +20,10 @@ class Product < ApplicationRecord
   has_many :orders, through: :line_items
   before_destroy :ensure_not_referenced_by_any_line_item
 
+  after_initialize :set_name_default
+  before_save :set_discount_price
```

Comment:
why diff callbacks for doing same sort of work?
learn about callbacks, and what callback should be used where

Reply:
done